src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -614,10 +614,13 @@ private static void parseSingleAltAllele(List<Allele> alleles, String alt, int l
     }
 
     public final static boolean canDecodeFile(final String potentialInput, final String MAGIC_HEADER_LINE) {
-        try {
-            return isVCFStream(new FileInputStream(potentialInput), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new GZIPInputStream(new FileInputStream(potentialInput)), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new BlockCompressedInputStream(new FileInputStream(potentialInput)), MAGIC_HEADER_LINE);
+        try(
+            final FileInputStream fis = new FileInputStream(potentialInput);
```

Comment:
Not your doing, and doesn't _need_ to be fixed, but this method is icky.  Why open up three streams (which will open three filehandles and do buffering) when you may only need the first one or two in many cases??